pkg/controller/hibernation/hibernation_controller_test.go

```diff
@@ -289,6 +293,158 @@ func TestReconcile(t *testing.T) {
 
 }
 
+func TestHibernateAfter(t *testing.T) {
+	logger := log.New()
+	logger.SetLevel(log.DebugLevel)
+
+	scheme := runtime.NewScheme()
+	corev1.AddToScheme(scheme)
+	batchv1.AddToScheme(scheme)
+	hivev1.AddToScheme(scheme)
+	hivev1.AddToScheme(scheme)
+
+	cdBuilder := testcd.FullBuilder(namespace, cdName, scheme).Options(
+		func(cd *hivev1.ClusterDeployment) {
+			cd.Spec.Installed = true
+			cd.Status.ClusterVersionStatus = &configv1.ClusterVersionStatus{
+				Desired: configv1.Update{
+					Version: "4.4.9",
+				},
+			}
+		},
+	)
+	o := clusterDeploymentOptions{}
+
+	tests := []struct {
+		name          string
+		setupActuator func(actuator *mock.MockHibernationActuator)
+		cd            *hivev1.ClusterDeployment
+
+		installedTime      time.Time
+		hibernateAfter     string
+		hibernateCondition hivev1.ClusterDeploymentCondition
+		expectRequeueAfter bool
+		expectedPowerState hivev1.ClusterPowerState
+	}{
+		{
+			name: "cluster due for hibernate no condition", // cluster that has never been hibernated and thus has no running condition
+			cd: cdBuilder.GenericOptions(testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "8h")).Options(
+				testcd.InstalledTimestamp(time.Now().Add(-10 * time.Hour))).Build(),
```

Comment:
```suggestion
			cd: cdBuilder.GenericOptions(testgeneric.WithAnnotation(constants.HibernateAfterAnnotation, "8h")).Build(
				testcd.InstalledTimestamp(time.Now().Add(-10 * time.Hour))),
```

Reply:
My expectation was that this change would be made for all of the test cases and not just the one that I called out as an example.